src/uct/rocm/base/rocm_base.c

```diff
@@ -292,9 +292,43 @@ int uct_rocm_base_is_dmabuf_supported()
     int dmabuf_supported = 0;
 
 #if HAVE_HSA_AMD_PORTABLE_EXPORT_DMABUF
-    dmabuf_supported = 1;
-#endif
+    FILE *fp             = NULL;
+    char kernel_opt1[28] = "CONFIG_DMABUF_MOVE_NOTIFY=y";
+    char kernel_opt2[20] = "CONFIG_PCI_P2PDMA=y";
+    int found_opt1       = 0;
+    int found_opt2       = 0;
+    struct utsname utsname;
+    char kernel_conf_file[128];
+    char buf[256];
+
+    if (uname(&utsname) == -1) {
+        ucs_trace("could not get kernel name");
+        goto exit;
+    }
+
+    ucs_snprintf_safe(kernel_conf_file, 128, "/boot/config-%s",
```

Comment:
seems quite hacky.. isn't there a better way to check it? such as query rocm driver? or the running kernel?
not always the config file exists in that location..

Reply:
I chatted with somebody from our kernel team, he was not aware of an easier solution. Furthermore, I did check on a couple of platforms that I have access to, including ubuntu, rhel, and sles systems, and the file did exist in all of them. So unless we have a better idea, I would suggest that we go ahead with this version for now. We can also adjust it later if a new/better solution comes up.

Reply:
Can we try obtaining dmabuf fd from room driver for a dummy allocation and check if it works?

Reply:
Unfortunately not. Here is the actual problem: even without these two kernel config options, the runtime will return a valid dmabuf handle that works if you register it with ibv_reg_dmabuf_mr, and as long as the buffer would be used only for this, you would be good. The problem occurs once the same buffer is also registered and used for IPC communication. I am do not fully know what is happening on the runtime/kernel level, but once a dmabuf handle has been created the IPC mechanism doesn't work on that buffer anymore, unless these two kernel option have been enabled.

Reply:
Is this something we can test in this function? i.e creating dmabuf then using the buffer with IPC

Reply:
I have testcodes where this being done (although they are multi-node tests, I didn't try whether importing the ipc handle on the same process as it was created on triggers the same problem). I don't think it is however practical to do this here. The problem appears if you import an ipc key *after* calling ibv_reg_dmabuf_mr function for the same buffer. For this I would have to go through the entire chain of required ibv functions since ibv_reg_dmabuf_mr requires a protection domain as an argument  I don't think that is a practical solution, and would in my opinion be a worse abstraction violation than checking for the two kernel config parameters in the rocm component.

Reply:
ok